model.py

```diff
@@ -9589,6 +9597,18 @@ def by_protocol(cls, _db, protocol):
             ).filter(ExternalIntegration.protocol==protocol)
         return qu
 
+    @classmethod
+    def by_datasource(cls, _db, data_source):
+        if isinstance(data_source, DataSource):
```

Comment:
A docstring on this method would be helpful -- I know what happens inside but it's not clear from looking at the implementation why you would call it.